File: python/knn_example.py
# python/knn_example.py
from sklearn.neighbors import KNeighborsClassifier
from sklearn.datasets import load_iris
from sklearn.model_selection import train_test_split

# Load sample data
X, y = load_iris(return_X_y=True)
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

# Train KNN (k=3)
knn = KNeighborsClassifier(n_neighbors=3)
knn.fit(X_train, y_train)

# Predict and score
score = knn.score(X_test, y_test)
print(f"Test accuracy: {score:.3f}")

File: r/knn_example.R
# r/knn_example.R
library(class)
data(iris)

set.seed(123)
train_idx <- sample(1:nrow(iris), 0.8 * nrow(iris))
train <- iris[train_idx, ]
test <- iris[-train_idx, ]

pred <- knn(train[, 1:4], test[, 1:4], train$Species, k = 3)
accuracy <- mean(pred == test$Species)
cat("Test accuracy:", round(accuracy, 3), "\n")